dive_core/tests/perf_metrics_data_test.cpp

```diff
@@ -0,0 +1,156 @@
+    EXPECT_THAT(record1, PerfMetricsRecordEq(expected_record1));
+    ASSERT_EQ(record1.m_metric_values.size(), 2);
+    EXPECT_EQ(std::get<int64_t>(record1.m_metric_values[0]), 123);
+    EXPECT_FLOAT_EQ(std::get<float>(record1.m_metric_values[1]), 1.23f);
+
+    // Check second record
+    Dive::PerfMetricsRecord expected_record2;
+    expected_record2.m_context_id = 2;
+    expected_record2.m_process_id = 200;
+    expected_record2.m_frame_id = 2000;
+    expected_record2.m_cmd_buffer_id = 20000;
+    expected_record2.m_draw_id = 2;
+    expected_record2.m_draw_label = 2;
+    expected_record2.m_program_id = 2;
+    expected_record2.m_draw_type = 2;
+    expected_record2.m_lrz_state = 2;
+    const auto& record2 = records[1];
+    EXPECT_THAT(record2, PerfMetricsRecordEq(expected_record2));
+    ASSERT_EQ(record2.m_metric_values.size(), 2);
+    EXPECT_EQ(std::get<int64_t>(record2.m_metric_values[0]), 456);
+    EXPECT_FLOAT_EQ(std::get<float>(record2.m_metric_values[1]), 4.56f);
+
+    const auto& metric_names = perf_metrics_data->GetMetricNames();
+    ASSERT_EQ(metric_names.size(), 2);
+    EXPECT_EQ(metric_names[0], "COUNTER_A");
+    EXPECT_EQ(metric_names[1], "COUNTER_B");
+
+    const auto& metric_infos = perf_metrics_data->GetMetricInfos();
+    ASSERT_EQ(metric_infos.size(), 2);
+    ASSERT_NE(metric_infos[0], nullptr);
+    EXPECT_EQ(metric_infos[0]->m_metric_id, 1);
+    ASSERT_NE(metric_infos[1], nullptr);
+    EXPECT_EQ(metric_infos[1]->m_metric_id, 2);
+}
+
+TEST_F(PerfMetricsDataTest, LoadFromCsvMalformedRowSkipped)
+{
+    auto
+    perf_metrics_data = Dive::PerfMetricsData::LoadFromCsv(TEST_DATA_DIR
+                                                           "/mock_perf_metrics_data_malformed.csv",
+                                                           *m_available_metrics);
+    ASSERT_TRUE(perf_metrics_data);
+    const auto& records = perf_metrics_data->GetRecords();
+    ASSERT_EQ(records.size(), 1);
+}
+
+TEST_F(PerfMetricsDataTest, LoadFromCsvFailedWithNoHeader)
+{
+    auto
+    perf_metrics_data = Dive::PerfMetricsData::LoadFromCsv(TEST_DATA_DIR
+                                                           "/mock_perf_metrics_data_no_header.csv",
+                                                           *m_available_metrics);
+    ASSERT_FALSE(perf_metrics_data);
+}
+
+TEST_F(PerfMetricsDataTest, LoadFromCsvWrongColumnsSkipped)
+{
+    auto perf_metrics_data = Dive::PerfMetricsData::
+    LoadFromCsv(TEST_DATA_DIR "/mock_perf_metrics_data_wrong_columns.csv", *m_available_metrics);
+    ASSERT_TRUE(perf_metrics_data);
+    const auto& records = perf_metrics_data->GetRecords();
+    ASSERT_EQ(records.size(), 0);
+}
+
+TEST(PerfMetricsData, LoadFromCsvUnknownMeticTypeSkipped)
+{
+    auto available_metrics = Dive::AvailableMetrics::LoadFromCsv(
+    TEST_DATA_DIR "/mock_available_metrics_unknown_type.csv");
+    ASSERT_TRUE(available_metrics);
+
+    auto perf_metrics_data = Dive::PerfMetricsData::
```

Comment:
Can omit the namespace since the test code is in the Dive namespace
```suggestion
    auto perf_metrics_data = PerfMetricsData::
```